### ISSUE TITLE
Arbitrary hex color values entered not working in UI (ColorBox.vue)

### ISSUE THREAD

**Issue by brymon68** [GoXLR-on-Linux/goxlr-ui#5]
Currently, in order to change the colors of various lighting components within the UI that ships with the goxlr-utility, the user has to select the correct color via the colorpicker component. However, if you enter a hex value (at least in the Mixer 'Mute' field) such as: ```#00FFFF```, the value does not update the preview color or update the color on the actual goxlr hardware. 

Now my understanding of vue is not great (I have alot of experience in React), but looking at this bit of code: https://github.com/GoXLR-on-Linux/goxlr-ui/blob/main/src/components/sections/lighting/ColourBox.vue#L79-L88

```
updateColour(event) {
      let value = event.target.value;
      const regex = /^#([a-f0-9]{6})\b$/
      if (value.match(regex)) {
        //this.$refs.colourInput.value = value;
        this.$emit('colour-changed', this.id, this.textValue);
      }
    }
  },
  ```
  
There are a couple issues: 

1. The regex is not sufficient as it does not account for uppercase characters. Should probably be: 
```  const regex = /^#([a-fA-F0-9]{6})\b$/```

2. I dont know too much about what the above event emitter is doing in the updateColour function: 

```this.$emit('colour-changed', this.id, this.textValue);```

But perhaps should be replaced by: 

```this.updateText(value)```
Replacing `this.$emit('colour-changed', this.id, this.textValue);` to `this.updateText(value)` in my local environment fixed the issue and accurately sets the correct color both in the preview but also on the GoXLR itself. I'm happy to push a PR but wanted to get guidance on what the this.$emit() call is doing. 

Thank you for your wonderful work. I am finally able to use GoXLR on mac and life is amazing :)

**Comment by FrostyCoolSlug** [comment]
That's fine to have as a PR, a valid hex is supposed to change the local textValue variable before the emit, which I missed or lost somewhere along the way, given that `updateText` does both, might as well recycle it.

As far as emits in Vue, they're simply events sent to the parent component. So for example in `LightingEffects.vue` the `<ColourBox />` component tags will have an attribute `@colour-changed="functionName"` which catches the triggered `colour-changed` emit and calls the function named with the parameters `(id, value)` representing the `this.id, this.textValue` values in the original emit.

So as long as the update is still being triggered in `updateText` (which I believe it is), it's perfectly fine to trigger there instead :)

**Comment by brymon68** [comment]
Created PR here: #6

**Comment by FrostyCoolSlug** [comment]
Thanks, I've pulled this in and it'll be available in 0.9.0
